src/java.base/share/classes/sun/security/ssl/SSLSessionContextImpl.java

```diff
@@ -69,6 +73,10 @@ final class SSLSessionContextImpl implements SSLSessionContext {
     private int cacheLimit;             // the max cache size
     private int timeout;                // timeout in seconds
 
+    private int currentKeyID;           // RFC 5077 session ticket key name
+    private final Map<Integer,          // Maps session keys to session state
+            SessionTicketExtension.StatelessKey> keyHashMap = new ConcurrentHashMap<>();
+
```

Comment:
It may be not necessary to allocate the map for client session context.  It may be fine to me the map allocation to constructor.

```
private final Map<Integer, SessionTicketExtension.StatelessKey> keyHashMap;
SSLSessionContextImpl(boolean server) {
    ....
    if (server) {
        keyHashMap = ...
        currentKeyID = ...
    } else {
        keyHashMap = empty map
    }
}
```